fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -660,16 +668,39 @@ public Stream<ComputeNode> backendAndComputeNodeStream() {
 
     public List<Long> seqChooseBackendIdsByStorageMedium(int backendNum, boolean needAvailable, boolean isCreate,
                                                          TStorageMedium storageMedium) {
-        final List<Backend> backends =
-                getBackends().stream().filter(v -> !v.diskExceedLimitByStorageMedium(storageMedium))
-                        .collect(Collectors.toList());
-        return seqChooseBackendIds(backendNum, needAvailable, isCreate, backends);
+
+        return seqChooseBackendIds(backendNum, needAvailable, isCreate,
+                v -> !v.diskExceedLimitByStorageMedium(storageMedium));
     }
 
     public List<Long> seqChooseBackendIds(int backendNum, boolean needAvailable, boolean isCreate) {
-        final List<Backend> backends =
-                getBackends().stream().filter(v -> !v.diskExceedLimit()).collect(Collectors.toList());
-        return seqChooseBackendIds(backendNum, needAvailable, isCreate, backends);
+
+        return seqChooseBackendIds(backendNum, needAvailable, isCreate, v -> !v.diskExceedLimit());
+    }
+
+    private List<Long> seqChooseBackendIds(int backendNum, boolean needAvailable, boolean isCreate,
+                                           Predicate<? super Backend> predicate) {
+
+        final List<Backend> wholeAliveBackends = getAvailableBackends();
+        if (CollectionUtils.isEmpty(wholeAliveBackends)) {
+            LOG.info("failed to find any alive backend");
+            return Collections.emptyList();
+        }
```

Comment:
available backend means alive and non-decommissioned backend, i think we need to check `needAvailable` here, how about changing the code like this,
```suggestion
        final List<Backend> candidateBackends = needAvailable ? getAvailableBackends() : getBackends();
        if (CollectionUtils.isEmpty(candidateBackends)) {
            LOG.info("failed to find any backend, needAvailable={}", needAvailable);
            return Collections.emptyList();
        }
```

and in this case, we can remove the `needAvailable` parameter and related checking logic in method
```
public synchronized List<Long> seqChooseBackendIds(int backendNum, boolean needAvailable, boolean isCreate, final List<Backend> srcBackends) { }
```
below.